app/models/setting/puppetdb.rb

```diff
@@ -20,14 +22,16 @@ def self.default_settings
     default_ssl_ca_file ||= (SETTINGS[:ssl_ca_file]).to_s
     default_ssl_certificate ||= (SETTINGS[:ssl_certificate]).to_s
     default_ssl_private_key ||= (SETTINGS[:ssl_priv_key]).to_s
+    default_api_version ||= 4
```

Comment:
I think, adding a db migration might also work for this. Let me know, if you want me to try that out.